src/FSharp.Configuration/ProvidedTypes.fs

```diff
@@ -941,7 +942,9 @@ type ProvidedSymbolType(kind: SymbolKind, args: Type list) =
         | SymbolKind.Array _ -> typeof<System.Array>
         | SymbolKind.Pointer -> typeof<System.ValueType>
         | SymbolKind.ByRef -> typeof<System.ValueType>
-        | SymbolKind.Generic gty  -> ProvidedSymbolType.convType args gty.BaseType
+        | SymbolKind.Generic gty  -> 
+            if gty.BaseType = null then null else
+            ProvidedSymbolType.convType args gty.BaseType
```

Comment:
This is just as it's intended to be used from the very beginning :) Every type provider must have it's own copy of these two files. The single central place where they should be changed is FSharp.TypeProviders.StarterPack. If you need deeper explanation, ask @dsyme.